source/postprocess/visualization/mass_error_face_integral.cc

```diff
@@ -0,0 +1,153 @@
+                cell_volume += fe_values.JxW(q);
+
+              double div_rho_u = 0.0;
+
+              for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+                {
+                  fe_face_values.reinit(cell,f);
+
+                  fe_face_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
+                      in.velocity);
+                  fe_face_values[this->introspection().extractors.temperature].get_function_values (this->get_solution(),
+                      in.temperature);
+                  fe_face_values[this->introspection().extractors.pressure].get_function_values (this->get_solution(),
+                                                                                                 in.pressure);
+                  fe_face_values[this->introspection().extractors.pressure].get_function_gradients (this->get_solution(),
+                      in.pressure_gradient);
+                  for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                    fe_face_values[this->introspection().extractors.compositional_fields[c]].get_function_values(this->get_solution(),
+                        composition_values[c]);
+
+                  in.position = fe_face_values.get_quadrature_points();
+
+                  in.strain_rate.resize(0);
+
+                  for (unsigned int i=0; i<n_q_points_face; ++i)
+                    {
+                      for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                        in.composition[i][c] = composition_values[c][i];
+                    }
+                  in.cell = &cell;
+
+                  this->get_material_model().evaluate(in, out);
+
+                  for (unsigned int q = 0; q < n_q_points_face; ++q)
+                    div_rho_u  += out.densities[q] * (in.velocity[q] * fe_face_values.normal_vector(q))
+                                  * fe_face_values.JxW(q);
+                }
+
+              (*return_value.second)(cell_index) = div_rho_u / cell_volume;
+            }
+
+        return return_value;
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(MassErrorFaceIntegral,
+                                                  "mass error face integral",
+                                                  "A postprocessor that computes some statistics about the "
+                                                  "error of the mass conservation equation. More precisely, it "
+                                                  "computes the norm and the integral of the equation "
+                                                  "$\\rho \\Nabla \\mathbf u + \\Nabla \\rho \\cdot \\mathbf u$. "
+                                                  "Therefore, its results indicate how accurate the chosen approximation "
```

Comment:
same here regarding the comments I have above. should document the fact that you compute *average* values.